lib/util/common.ts

```diff
@@ -59,3 +61,32 @@ export function detectPlatform(
 
   return null;
 }
+
+export function parseJson(content: string | null, filename: string): unknown {
+  if (!content) {
+    return null;
+  }
+
+  return filename.endsWith('.json5')
+    ? JSON5.parse(content)
+    : parseJsonWithFallback(content, filename);
+}
+
+export function parseJsonWithFallback(
+  content: string,
+  context: string
+): unknown {
+  let parsedJson: unknown;
+
+  try {
+    parsedJson = JSON.parse(content);
+  } catch (err) {
+    parsedJson = JSON5.parse(content);
+    logger.warn(
+      { context },
+      'File contents are invalid JSON but parse using JSON5. Support for this will be removed in a future release so please change to a support .json5 file name or ensure correct JSON syntax.'
```

Comment:
This log message is a bit confusing. How about something like this:

> File contents are invalid JSON but parse corectly as JSON5. In a future release, Renovate will strictly parse JSON as JSON, and JSON5 as JSON5. To prevent breaking your Renovate config in a upcoming major release: please make sure the file extension matches the content syntax.

In any case, we should be clear about:

- Current behavior (Renovate silently eats wrong file format)
- Upcoming behavior (Renovate will parse strictly according to file extension)
- What the user most do to prevent future breakage (make sure file extension matches file content syntax)